pensieve/pre_treatment.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+import attr
+from pandas import DataFrame
+import re
+
+
+@attr.s(auto_attribs=True)
+class PreTreatment(ABC):
+    """
+    Represents an abstract pre-treatment step applied to data before
+    calculating statistics.
+    """
+
+    @classmethod
+    def name(cls):
+        """Return snake-cased name of the statistic."""
+        return re.sub(r"(?<!^)(?=[A-Z])", "_", cls.__name__).lower()
+
+    @abstractmethod
+    def apply(self, df: DataFrame) -> DataFrame:
```

Comment:
I don't think it's right to apply these to the whole DataFrame. These should know what columns they're operating on, so let's accept a column name as an argument.